kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/storageApi/StorageWriteApiDefaultStream.java

```diff
@@ -0,0 +1,278 @@
+                        if (!tableCreationOrUpdateAttempted) {
+                            logger.info("Attempting to create table {} ...", tableName);
+                            tableCreationOrUpdateAttempted = true;
+                            attemptTableCreation(TableNameUtils.tableId(table), getSinkRecords(rows));
+                            // Table takes time to be available for after creation
+                            additionalWait = 30000; // 30 seconds;
+                            additionalRetries = 30;
+                            logger.info("Table creation completed {} ...", tableName);
+                        }
+                    } else if (!BigQueryStorageWriteApiErrorResponses.isRetriableError(e.getMessage())) {
+                        logger.error("Failed to create Default stream writer on table {}", tableName);
+                        throw new BigQueryStorageWriteApiConnectException("Failed to create Default stream writer on table " + tableName, e);
+                    }
+                    logger.warn("Failed to create Default stream writer on table {} due to {}. Retry attempt {}...", tableName, e.getMessage(), attempt);
+                    mostRecentException = e;
+                    attempt++;
+                }
+            } while (attempt < (retry + additionalRetries));
+            throw new BigQueryStorageWriteApiConnectException(
+                    String.format(
+                            "Exceeded %s attempts to create Default stream writer on table %s ",
+                            (retry + additionalRetries), tableName
+                    ), mostRecentException);
+
+        });
+    }
+
+    /**
+     * Calls AppendRows and handles exception if the ingestion fails
+     *
+     * @param tableName  The table to write data to
+     * @param rows       The records to write
+     * @param streamName The stream to use to write table to table. This will be DEFAULT always.
+     */
+    @Override
+    public void appendRows(TableName tableName, List<Object[]> rows, String streamName) {
+        JSONArray jsonArr = new JSONArray();
+
+        for (Object[] item : rows) {
+            jsonArr.put(item[1]);
+        }
+
+        logger.debug("Sending {} records to write Api default stream on {} ...", rows.size(), tableName);
+        int attempt = 0;
+        Exception mostRecentException = null;
+        boolean tableCreationOrUpdateAttempted = false;
+        int additionalRetries = 0;
+        int additionalWait = 0;
+        do {
+            try {
+                if (attempt > 0) {
+                    waitRandomTime(additionalWait);
+                }
+                logger.trace("Sending records to Storage API writer...");
+                JsonStreamWriter writer = getDefaultStream(tableName, rows);
+                ApiFuture<AppendRowsResponse> response = writer.append(jsonArr);
+                AppendRowsResponse writeResult = response.get();
```

Comment:
response.get() would block the thread. Should we make it return `CompletableFuture` instead?

Reply:
Th execution of this part is already delegated to a thread pool. This would not block the main thread.

Reply:
but this would block thread in the threadpool. Would we be able to use this under heavy load? I am thinking under heavy traffic, this would cause frequent flow control because threads may be busy waiting for response.

Reply:
Yes. We are yet to do performance testing but the primary reasons for choosing this approach is Non-blocking call handling did not fit in well with retry mechanism and exception handling design we had. This has been tested on average load and works fine. For special cases where load is high, there are configs which can be tweaked to keep the connector performant.  Also we have batch mode which should cover for it.

Reply:
We have usecase where we load high traffic of data through storage write api. We don't want to opt for batch load as it does not provide SLA gurantee. Can we look into resiliency4J library for async retry?

Reply:
Thanks for the inputs @SubhamSinghal. We will do adequate performance testing on our end to verify the limits. 
Also, we will consider looking into the suggested library.